x-pack/libbeat/management/managerV2_test.go

```diff
@@ -584,6 +586,16 @@ func TestFlattenedDataStreams(t *testing.T) {
 				"data_stream.namespace": expectedNamespace,
 				"data_stream.type":      expectedType,
 			}),
+			// If the data_stream is not flattened, Beats will receive
+			// it twice, in the Config.Source and as DataStream, that is the
+			// normal behaviour when the Elastic-Agent is managed by Fleet.
+			// So we also add the DataStream here to make sure there will be
+			// no conflict.
```

Comment:
Is this because the `DataStream` object from the Elastic Agent doesn't parse the flattened variant? There is no reason for these two cases to be different in the protocol.

It's good to defend against this problem, but the protocol should be consistent or other developers are going to have this exact same bug.

I think we should follow up and fix this in the agent as well so that there is no inconsistency here.

Reply:
> Is this because the DataStream object from the Elastic Agent doesn't parse the flattened variant? 

Yes. If I recall correctly, the Agent parses it directly into the Protobuf type that does not support the flattened syntax.

>It's good to defend against this problem, but the protocol should be consistent or other developers are going to have this exact same bug.
>
>I think we should follow up and fix this in the agent as well so that there is no inconsistency here.

Should we then revert the original commit supporting flattened keys on Beats and re-implement it on Elastic-Agent?

Reply:
> Should we then revert the original commit supporting flattened keys on Beats and re-implement it on Elastic-Agent?

Yes, we need to change the agent as well or every input is going to need to handle these two cases which is definitely result in bugs.